### ISSUE TITLE
Publish to npm

### ISSUE THREAD

**Issue by kbjr** [dmauro/Keypress#83]
Can we get this up on npm?

**Comment by gbmoretti** [comment]
:+1:

**Comment by browniefed** [comment]
Yes, this would be extremely beneficial

**Comment by dmauro** [comment]
Will do. Out of curiosity, how exactly do you all plan on using this w/ Node?

**Comment by browniefed** [comment]
@dmauro just using NPM as a dependency manager, and then webpack to bundle everything (or browserify)

**Comment by maxkorp** [comment]
Yes plz. Working on an atom-shell app, so I like to manage all my deps with NPM, as atom-shell lets us just require modules from node_modules just as if it were true node (even on the client). I also use npm with webpack/browserify for static websites, just to avoid having both npm AND bower (deps becomes client deps, at that point).

**Comment by maxkorp** [comment]
As a note, it appears @kbjr has already done this (https://github.com/UmbraEngineering/keypress-npm) but it would still be nice to have this straight from the source.

**Comment by dmauro** [comment]
Sorry I've dropped the ball on this. Been very busy at work lately. I'll see if I can manage it this weekend.

**Comment by derhuerst** [comment]
Any news on this? Any other library you recommend?

**Comment by dmauro** [comment]
Sorry I pushed an update and forgot I wanted to sneak this into that update. Will try to find some time soon to make another release for this. Thanks for keeping on top of me about it.

**Comment by dev-tim** [comment]
+1

**Comment by th3fallen** [comment]
👍  if you dont mind i can publish it for you? it literally takes two seconds but would be amazing to have it up there

**Comment by bertofer** [comment]
+1

 I plan to use it in an Electron app, and npm is the only package manager I would like to keep

**Comment by dmauro** [comment]
Yeah sorry, I was trying to get this working and ran into a bug with npm. I'll give this another shot.

**Comment by dhensche** [comment]
Any word on this? I was hoping to use this library in a project, but I can only include new dependencies that are retrievable via npm

**Comment by dmauro** [comment]
I think I need to clean out old JS/NPM cruft on my machine or something. @th3fallen you mentioned you wouldn't mind handling it? I could really use the help. I need to take a minute to put a call for help on the main page too. I'm just not able to give this project the attention it needs anymore.

**Comment by th3fallen** [comment]
@dmauro yeah no worries i'll publish it now, and add you as an owner of it. if you need anything feel free to ask i'll lend a hand.

**Comment by th3fallen** [comment]
Actually @dmauro looks like @kbjr has already published a fork under the name keypress.js since there is already a package published named keypress if you could rename your package name in package.json to match keypress.js and we could sweet talk @kbjr to remove his listing we could publish this under that name.

**Comment by kbjr** [comment]
That's fine with me. Let me know if you plan to move forward, and I'll take it down when you're ready to put yours up.

**Comment by th3fallen** [comment]
@kbjr i'll submit a Pr here to update the package name now and ping you when it gets merged and we'll be good

**Comment by hustcer** [comment]
Need to be updated

**Comment by th3fallen** [comment]
@kbjr got the package updated you can drop your's whenever you have a moment.

**Comment by kbjr** [comment]
seems that npm's new policy prevents me from unpublishing, i'm going to contact them now.

**Comment by th3fallen** [comment]
@kbjr yeah no worries.

**Comment by kbjr** [comment]
Okay, email sent.

**Comment by kbjr** [comment]
Just got a reply back

> Based on recent changes to the unpublish policy, the keypress.js package doesn’t appear to be one that we would unpublish. I can, however, transfer the package to the @npm user account, deprecate it and publish a deprecation placeholder. That would remove it from your profile, alert users that it’s no longer supported, and change the output of npm owner ls keypress.js to be npm <npm@npmjs.com>.

> Alternatively, I could just transfer ownership of the package over to @th3fallen user account. They could then publish an update as needed.

I've told them to transfer ownership over to you. You should be able to publish a new version then.

**Comment by th3fallen** [comment]
Thanks for the co-operation @kbjr https://www.npmjs.com/package/keypress.js is now this repo, i've added @dmauro as a collaborator as well. we should be good to go.

**Comment by th3fallen** [comment]
whenever you get your npm stuff sorted out let me know and i'll transfer ownership to you @dmauro

**Comment by hustcer** [comment]
`import Keypress from 'keypress.js';` in es6 not work. It works before. There should be a `keypress.js` file in the root directory

**Comment by kmarwood** [comment]
@hustcer Yes. Either that or "main": "keypress.js" in package.json needs to change to "main": "keypress-2.1.4.min.js" or similar...
